apps/frontend/src/components/global/checklist/ChecklistRulesTable.vue

```diff
@@ -0,0 +1,231 @@
+          </v-chip>
+        </template>
+        <template #[`item.ruleVersion`]="{item}">
+          {{ shortStigId(item.ruleVersion) }}
+        </template>
+        <template #[`item.ruleId`]="{item}">
+          {{ shortRuleId(item.ruleId) }}
+        </template>
+        <template #[`item.cciRef`]="{item}">
+          {{ shortRuleId(item.cciRef) }}
+        </template>
+      </v-data-table>
+    </v-card-text>
+  </v-card>
+</template>
+
+<script lang="ts">
+import {ChecklistFilter, FilteredDataModule} from '@/store/data_filters';
+import {FileID} from '@/store/report_intake';
+import {ChecklistVuln} from '@mitre/hdf-converters';
+import {InspecDataModule} from '@/store/data_store';
+import _ from 'lodash';
+import {Component, Prop, Vue} from 'vue-property-decorator';
+
+@Component
+export default class ChecklistRulesTable extends Vue {
+  @Prop({type: Object, required: true}) readonly allFilter!: ChecklistFilter;
+  @Prop({type: Array, required: true}) readonly fileFilter!: FileID[];
+  @Prop({type: Boolean, required: true}) readonly shortIdEnabled!: boolean;
+  @Prop({type: Array, required: true}) readonly rules!: ChecklistVuln[];
+
+  shortRuleId(ruleId: string) {
+    if (this.shortIdEnabled) return ruleId.split('r')[0] || ruleId;
+    else return ruleId;
+  }
+
+  shortStigId(stigId: string) {
+    if (this.shortIdEnabled) return stigId.split('-').slice(0, 2).join('-');
+    else return stigId;
+  }
+
+  statusColor(status: string | undefined) {
+    switch (status) {
+      case 'Passed':
+        return 'statusPassed';
+      case 'Not Applicable':
+        return 'statusNotApplicable';
+      case 'Failed':
+        return 'statusFailed';
+      default: // Not_Reviewed
+        return 'statusNotReviewed';
+    }
+  }
+
+  tableItems: ChecklistVuln[] = [];
+  numItems = 0;
+  getFiltered(rules: ChecklistVuln[]) {
+    this.tableItems = rules;
+    this.numItems = this.tableItems.length;
+  }
+
+  numStatus(status: string): string {
+    return this.tableItems
+      .filter((item) => item.status === status)
+      .length.toString();
+  }
+
+  shortStatus(status: string) {
+    if (this.shortIdEnabled) {
+      switch (status) {
+        case 'Not Reviewed':
+          return 'NR';
+        case 'Failed':
+          return 'F';
+        case 'Passed':
+          return 'P';
+        case 'Not Applicable':
+          return 'NA';
+      }
+    }
+    return status;
+  }
+
+  numSeverity(severity: string): string {
+    return this.tableItems
+      .filter((item) => item.severity === severity.toLowerCase())
+      .length.toString();
+  }
+
+  getChecklist(fileID: FileID[]) {
+    return InspecDataModule.allChecklistFiles.find(
+      (f) => f.uniqueId === fileID[0]
+    );
+  }
+
+  showRule(rule: ChecklistVuln) {
+    FilteredDataModule.selectRule(rule);
+  }
+
+  checkSelected(rule: ChecklistVuln) {
+    if (rule.ruleId === FilteredDataModule.selectedRule.ruleId)
+      return 'selectedRow';
+  }
+
+  get headers() {
```

Comment:
review this function cause i think the implementation can be considerably simplified esp with the filtering changes no longer requiring the hidden row thing to happen

Reply:
Simplified